launcher/settingsView/csettingsview_moc.cpp

```diff
@@ -107,6 +112,33 @@ CSettingsView::CSettingsView(QWidget * parent)
 {
 	ui->setupUi(this);
 
+	// TODO: read available resolutions from mods
+	// please keep this list sorted
+	const auto supportedResolutions = {
+		QSize
+		{1024, 600},
+		{1024, 768},
+		{1181, 664},
+		{1280, 720},
+		{1280, 768},
+		{1280, 800},
+		{1280, 960},
+		{1280, 1024},
+		{1366, 768},
+		{1440, 900},
+		{1600, 1200},
+		{1680, 1050},
+		{1920, 1080},
+	};
+	// TODO: handle multiple screens
+	const auto screenSize = qGuiApp->primaryScreen()->size();
+	for(const auto & resolution : supportedResolutions)
+	{
+		if(screenSize.width() < resolution.width() || screenSize.height() < resolution.height())
+			break;
+		ui->comboBoxResolution->addItem(sizeToString(resolution));
+	}
```

Comment:
Let's remove implicit dependency on index as it can create unexpected bugs with missing resolutions. Do something like this instead
```suggestion
	for(int rId = 0; rId < supportedResolutions.size(); ++rId)
	{
		if(screenSize.width() < supportedResolutions[rId].width() || screenSize.height() < supportedResolutions.height())
			continue;
		ui->comboBoxResolution->addItem(sizeToString(supportedResolutions[rId]), rId);
	}
```

Don't commit this as is, because then you need to extract UserData from item

Reply:
not sure I'm following. When the loop is entered, the combox has only one hardcoded value: 800x600. The list of resolutions was previously hardcoded in the .ui file, now it's in .cpp with the filter.

Reply:
filter cuts off everything since one dimension higher than resolution. I suggest to use "continue" instead

Reply:
that's why the hardcoded list is sorted and respective comment is given :)

it's kind of temporary workaround anyway. The proper way is to obtain resolutions from mods.

Reply:
I'll give you an example: If I have device with screen 1366x768 you code won't give me option to select proper resolution, because cycle will break on 1280x800 (screenSize.height < resolution.height)

I believe it's a bug which needs to be fixed prior merge.

Reply:
Thank you for the example! Now I got it. You're absolutely right.